plugins/com.google.cloud.tools.eclipse.appengine.localserver/src/com/google/cloud/tools/eclipse/appengine/localserver/ui/StaleResourcesStatusHandler.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright 2017 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.eclipse.appengine.localserver.ui;
+
+import com.google.cloud.tools.eclipse.appengine.localserver.Messages;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.core.runtime.Status;
+import org.eclipse.debug.core.ILaunchConfiguration;
+import org.eclipse.debug.core.IStatusHandler;
+import org.eclipse.debug.internal.ui.DebugUIPlugin;
+import org.eclipse.debug.ui.DebugUITools;
+import org.eclipse.jface.dialogs.MessageDialog;
+import org.eclipse.swt.widgets.Shell;
+
+/**
+ * A simple prompter for the Platform/Debug framework to prompt the user to confirm whether the
+ * launch should continue when possibly stale resources have been found.
+ */
+public class StaleResourcesStatusHandler implements IStatusHandler {
+  /** The error code indicating that there may be stale resources. */
+  static final int STALE_RESOURCES_CODE = 255;
+
+  /** The status object to pass into the debug prompter */
+  public static final IStatus CONTINUE_LAUNCH_REQUEST = new Status(IStatus.INFO,
+      "com.google.cloud.tools.eclipse.appengine.localserver", STALE_RESOURCES_CODE, "", null);
+
+  @Override
+  public Object handleStatus(IStatus status, Object source) throws CoreException {
+    if (source instanceof ILaunchConfiguration) {
+      ILaunchConfiguration config = (ILaunchConfiguration) source;
+      if (DebugUITools.isPrivate(config)) {
+        return Boolean.FALSE;
```

Comment:
From what I gather, it's used for launches that aren't visible to the user.  Like Ant builds or Maven builds.  We could conceivably usr launches for our staging, etc.  The advantage being that it's tracked and managed by the launch info.